### ISSUE TITLE
ezgmail needs manual authentication once a week

### ISSUE THREAD

**Issue by nemethpetergyor** [asweigart/ezgmail#48]
I have been using ezgmail on a raspberry pi 4 for a month now to automatically check and download e-mail attachments every 10 minutes and it's working flawlessly except for an annoying authentication issue. I think I set up the OAuth correctly (according to the [latest instructions](https://github.com/asweigart/ezgmail)), but after a week of operation with no issues the script asks me to manually do a new authentication:

```
Failed to start a local webserver listening on either port 8080 or port 8090. Please check your firewall settings and locally running programs that may be blocking or using those ports.

Falling back to --noauth_local_webserver and continuing with authorization.

Go to the following link in your browser:
```

After I'm doing so, the script starts to work for another week again.
Is it a limitation of ezgmail or am I missing something?

**Comment by nemethpetergyor** [comment]
Yes, those files are present. The python script that is using ezgmail runs every 11 minutes (scheduled with cron) with no issues for a week. After that I need to do the authentication again with a browser. The script is on a rPi4 running 24/7 (with no scheduled reboots).
Do you think that something is wrong with the token/credentials JSON files and ezgmail can work without them (with only the manual authentication) for a week?

**Comment by nemethpetergyor** [comment]
Thanks for your quick reply.
I don't call the init() function. I just import ezgmail and start using the search function.
Cron runs a bash script that first changes the directory (cd) to where the token+credentials JSON files are located together with the python script. Then it executes the script by giving the full absolute path of it to python3.
I don't think that it should work even once if ezgmail couldn't find the JSONs, right?

**Comment by this-duck** [comment]
Same problem here. Not passing the tokens directly, but doing it as per Al's instuctions (and as Author stated) by just importing ezgmail.

**Comment by nemethpetergyor** [comment]
I can confirm that the following steps solved this issue for me: Go to google cloud console under oauth2 consent screen and set it to production, then authorize it again with your refreshed token and it won't expire.

My issue when I first tried this was that I forgot to refresh/regenerate the token and I was also worried about the validation process (YouTube video, etc.) after putting the app into production state. However, now I know that you can leave it in a pending validation state indefinitely.
